packages/providers/src.ts/base-provider.ts

```diff
@@ -923,7 +929,7 @@ export class BaseProvider extends Provider implements EnsProvider {
         // only an external call for backends which can have the underlying
         // network change spontaneously
         const currentNetwork = await this.detectNetwork();
-        if (network.chainId !== currentNetwork.chainId) {
+        if (network.chainId && network.chainId !== currentNetwork.chainId) {
```

Comment:
I think that we should populate the `network.chainId` every time we create an instance of a provider. We know the CHAIN IDs for Previewnet, Testnet and Mainnet so we should be able to populate that right?

Reply:
Yes, this is a leftover from the network fields research, where I didn't pass the correct network. Will remove